src/Elastic.Apm.AspNetFullFramework/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Web;
+
+namespace Elastic.Apm.AspNetFullFramework.Extensions
+{
+	internal static class HttpRequestExtensions
+	{
+		//
+		// Implementation of "HasFormContentType" and its helper methods
+		// "HasApplicationFormContentType" and "HasMultipartFormContentType" took inspiration from
+		// https://source.dot.net/#Microsoft.AspNetCore.Http/Features/FormFeature.cs
+		//
```

Comment:
Hahh! I wanted to talk about this in the zoom call - but then I forgot it.

This is fine and we can definitely take this as is. 

At the same time, we try to honor all these licences. In order to do so, we should create a `NOTICE` file under the `Elastic.Apm.AspNetFullFramework` folder and put something like that into it (I just wrote this quickly, feel free to adapt it if needed):
```
==========
Notice for: Elastic.Apm.AspNetFullFramework
----------

The code in `src/Elastic.Apm.AspNetFullFramework/Extensions/HttpRequestExtensions.cs` is licensed under the MIT license by the .NET Foundation.

The MIT License (MIT)

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```

So 2 things here:
- We make it clear which file contains code from a 3rd party source
- The MIT license requires that we need to put the MIT license itself into "all copies or substantial portions of the Software", so we add the text of the license as well.

Also make sure the `NOTICE` file is packable - like [here](https://github.com/elastic/apm-agent-dotnet/blob/main/src/Elastic.Apm/Elastic.Apm.csproj#L101-L106).

You can see an example of this in https://github.com/elastic/apm-agent-dotnet/blob/main/src/Elastic.Apm/NOTICE. 

The result will be a `NOTICE` file in all our distributions and it'll list all code that we copied from somewhere according to the license of that given code.